src/server/scripts/Northrend/isle_of_conquest.cpp

```diff
@@ -152,28 +153,11 @@ class spell_ioc_gunship_portal : public SpellScriptLoader
             void HandleScript(SpellEffIndex /*effIndex*/)
             {
                 Player* caster = GetCaster()->ToPlayer();
-                /*
-                 * HACK: GetWorldLocation() returns real position and not transportposition.
-                 * ServertoClient: SMSG_MOVE_TELEPORT (0x0B39)
-                 * counter: 45
-                 * Tranpsort Guid: Full: xxxx Type: MOTransport Low: xxx
-                 * Transport Position X: 0 Y: 0 Z: 0 O: 0
-                 * Position: X: 7.305609 Y: -0.095246 Z: 34.51022 O: 0
-                 */
-                caster->TeleportTo(GetHitCreature()->GetWorldLocation(), TELE_TO_NOT_LEAVE_TRANSPORT);
-                /*
-                 * HACK: This aura should be added by 20212 and 20213 but can't find any SMSG_SPELL_GO. Could't find their position.
-                 * ServerToClient: SMSG_AURA_UPDATE (0x0072)
-                 * [0] CasterGUID: Full: xxxxx Type: Unit Entry: 20212 Low: xxx
-                 * [0] Flags: None (0)
-                 * [0] Caster Level: 60
-                 * [0] Spell ID: 66656
-                 * [0] Charges: 0
-                 * [0] Effect Mask: 1
-                 * [0] Slot: 37
-                 * Guid: Full: xxxxx Type: Player2 Low: xxxxx
-                 */
-                caster->AddAura(SPELL_PARACHUTE, caster);
+
+                if (GetHitCreature()->GetTransport())
+                    GetHitCreature()->GetTransport()->AddPassenger(caster);
+
+                caster->TeleportTo(GetHitCreature()->GetMapId(), GetHitCreature()->GetTransOffsetX(), GetHitCreature()->GetTransOffsetY(), GetHitCreature()->GetTransOffsetZ(), GetHitCreature()->GetTransOffsetO(), TELE_TO_NOT_LEAVE_TRANSPORT);
```

Comment:
Just tested it again, and got teleported to the gunship ^^, should I change it and make it like the player only gets teleported if there is a transport ?

Edit: found out that when the battleground ends I get teleported to the middle of nowhere